rand_distr/src/weighted_tree.rs

```diff
@@ -97,6 +96,13 @@ pub struct WeightedTreeIndex<
 impl<W: Clone + PartialEq + PartialOrd + SampleUniform + SubAssign<W> + Weight>
     WeightedTreeIndex<W>
 {
+    /// Creates a new [`WeightedTreeIndex`] without weights.
+    pub fn new_empty() -> Self {
+        Self {
+            subtotals: Vec::new(),
+        }
+    }
+
```

Comment:
This isn't reformatting. Would be nicer in a different PR.

Reply:
This is relevant to Clippy warnings. It suggests to replace `new(&[])` with `new([])`, but doing so results in compilation error. I could simply `allow` this lint instead for now.

Reply:
Yes, but Clippy is not `rustfmt`. It might be easiest just to start over with just the `rustfmt` stuff then merge the other parts of this PR later.